components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/approles/impl/AppAssociatedRolesResolverImpl.java

```diff
@@ -346,6 +415,14 @@ private List<String> getUserGroups(AuthenticatedUser authenticatedUser) throws A
         return userGroups;
     }
 
+    private boolean isSharedUserAccessingSharedOrg(AuthenticatedUser authenticatedUser) {
+
+        return StringUtils.isNotEmpty(authenticatedUser.getUserSharedOrganizationId()) &&
+                StringUtils.isNotEmpty(authenticatedUser.getAccessingOrganization()) &&
+                StringUtils.equals(authenticatedUser.getUserSharedOrganizationId(),
+                        authenticatedUser.getAccessingOrganization());
+    }
```

Comment:
We can remove this and add a new method
isOrganizationSwitchedUser()
if resident & accessing org are not equal, it is an organization switching.